[GTK][WPE] Fix GLDisplay.cpp build issue with gcc-14

nikolaszimmermann opened the pull request:
#### 787897787c4e3288b03486c2653d9327459f3a1d
<pre>
[GTK][WPE] Fix GLDisplay.cpp build issue with gcc-14
<a href="https://bugs.webkit.org/show_bug.cgi?id=297901">https://bugs.webkit.org/show_bug.cgi?id=297901</a>

Reviewed by Carlos Garcia Campos.

queryDMABufFormats() initializes a static variable, depending on a
passed argument &apos;supportsModifiers&apos;, which could change from invocation
to invocation -- that&apos;s not a good idea. Fix the problem, and avoid
gcc 14 complaining.

* Source/WebCore/platform/graphics/egl/GLDisplay.cpp:
(WebCore::queryDMABufFormats):

Canonical link: <a href="https://commits.webkit.org/299194@main">https://commits.webkit.org/299194@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/93a76d6a931cd8b1ff146e59e6e59c0b049990dc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118219 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37897 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28536 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124378 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70260 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120097 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38590 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46479 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89707 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59346 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121172 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30739 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105995 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70203 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29809 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24114 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68044 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100165 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24288 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127452 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45122 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34015 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98391 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45483 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102216 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98178 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24961 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43574 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21562 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41587 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44994 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50668 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44454 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47799 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46143 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ccd42c9d8f1d0b8629be9dd9e302a6a10baedf99. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49883

carlosgcampos reviewed (approved):
Thanks! Please, check my comments before landing.

nikolaszimmermann commented:
Hehe, yes, that's indeed much cleaner, stupid me.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9bddae81bdb8162022ccc0f0813fd2674a5fbe44. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49883

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/93a76d6a931cd8b1ff146e59e6e59c0b049990dc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49883

nikolaszimmermann committed:
[GTK][WPE] Fix GLDisplay.cpp build issue with gcc-14
https://bugs.webkit.org/show_bug.cgi?id=297901

Reviewed by Carlos Garcia Campos.

queryDMABufFormats() initializes a static variable, depending on a
passed argument 'supportsModifiers', which could change from invocation
to invocation -- that's not a good idea. Fix the problem, and avoid
gcc 14 complaining.

* Source/WebCore/platform/graphics/egl/GLDisplay.cpp:
(WebCore::queryDMABufFormats):

Canonical link: https://commits.webkit.org/299194@main

webkit-commit-queue commented:
Committed 299194@main (787897787c4e): <https://commits.webkit.org/299194@main>

Reviewed commits have been landed. Closing PR #49883 and removing active labels.
